ee/auth/saml/router.js

```diff
@@ -83,7 +83,7 @@ router.post(
       throw new Error('Missing one or more SAML attributes');
     }
 
-    const params = [authUid, authName, authUin, 'SAML'];
+    const params = [authUid, authName, authUin, 'SAML', institutionId];
```

Comment:
How secure is `req.params.institution_id`? Normally I would think it's untrusted user data. Can we verify it somehow.

I'm thinking of a case where a user is working with an evil IdP. The user gets the IdP to generate a signed auth message for `nathan@prairielearn.com` and then the user sends that to the route with `institution_id` for `prairieLearn.com`. PL blindly trusts the `institution_id` from the route and auths the user.

Reply:
I believe this is handled by signing: it would be impossible for the evil IdP to have signed a response that would correctly validate with the _real_ IdP's public certificate. Does that sound correct to you?

Reply:
But how do we know that the `institution_id`-from-route matches the `institution_id` in the signed response? When we call `passport.authenticate()` we don't pass it the `institution_id` from the route, we just process the SAML response. And then later we do `const institutionId = req.params.institution_id` but we don't check that against anything. Am I missing a checking step somewhere?

Reply:
I think you are (at least, I hope I'm not). We use `MultiSamlStrategy`, and the options for that include `getSamlOptions`, which will retrieve the details (including certs) based on `req.params.institution_id`:

https://github.com/PrairieLearn/PrairieLearn/blob/6bb8c1d183977ad61139a1e8cb0c88adfedd6af0/ee/auth/saml/index.js#L6-L10

Reply:
Ok, great, I buy that. Thanks for explaining!